src/model/topology.cpp

```diff
@@ -194,12 +194,14 @@ void Topology::Specs::ParseAccelergyERT(config::CompoundConfigNode ert)
       for (unsigned i = 0; i < NumNetworks(); i++) {
           // only check the user-defined networks
           auto networkSpec = GetNetwork(i);
-          if (networkSpec->Type() == "SimpleMulticast" && networkSpec->name == componentName){
-            assert(false); // FIXME: why is the name of a network class exposed in this file?
-            //if (std::static_pointer_cast<SimpleMulticastNetwork::Specs>(networkSpec)){
-            //  // std::cout << "simple multicast component identified: " << componentName << std::endl;
-            //  std::static_pointer_cast<SimpleMulticastNetwork::Specs>(networkSpec)->accelergyERT = componentERT;
-            //}
+          if (networkSpec->SupportAccelergyTables() && networkSpec->name == componentName){
```

Comment:
I like what you've done to clean up the API. One question is whether this API method should be applicable to *all* modules instead of just the network module? And is there a more generic name for the method or are you specifically looking for "AccelergyTables"?